src/twisted/persisted/aot.py

```diff
@@ -447,8 +447,6 @@ def unjellyAO(self, ao):
             else:
                 raise TypeError("Unsupported AOT type: %s" % t)
 
-        del self.stack[-1]
-
```

Comment:
This follows an if/else clause where both all branches return or raise.

Someone familiar with `twisted.persisted` may want to consider if popping the stack here should be happening, but it's not happening now.